### DISCUSSION TITLE
SimpleTTS output from A2DP

### DISCUSSION THREAD

**Discussion by d3mac123** [pschatzmann/arduino-audio-tools#987] [Q&A] [answered]
[bringing this discussion [https://github.com/pschatzmann/ESP32-A2DP/discussions/469](https://github.com/pschatzmann/ESP32-A2DP/discussions/469) to here, as suggested]

Phil,
In response to your question, I am familiar with coding for Arduino but never worked with hardware (the reason why the i2s is something I am not experienced with) nor with audio. For now, I guess I will generate/convert 10 Audacity audio files (representing 0 to 9 numbers) and use the bt_music_sender_write_8bit example (which is working with my headset). However, if someone (I know I have abused your time already) has some time to share a sample code of a TTS number outputting to the A2DPStream class properly, it would be great.

**Comment by pschatzmann** [comment]
Oh, actually I suggest to start with some reading: Take a look at the Wiki of this project.
If you don't have any I2S DAC, you can use the AnalogOutputStream and test with some headphones

The following topics are most relevant for your case:
- Introduction
- Converting the Format

You might also want to look at the available examples (e.g in the stream and tts subdirectories)

**Comment by pschatzmann** [comment, marked as answer]
I have added and tested [a corresponding example](https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-tts/streams-simple_tts-a2dp).

In order to make things work, I needed to extend the A2DPStream a bit and make sure that the resampling is not writing too many requests to it. You will need to upgrade to the latest version...

I was surprised that all of this fits in PSRAM:
Sketch uses 1706233 bytes (54%) of program storage space. Maximum is 3145728 bytes.

**Reply by d3mac123** [reply]
So far what I found (thanks by the way):
- I removed the old library and installed the new ESP32-A2DP library;
- Running your example I got this error: 
```
/private/var/folders/j0/_j5h8rw161l2lrx2khy0_2wh0000gn/T/.arduinoIDE-unsaved202387-31211-6pwcpw.qowm7/sketch_sep7a/sketch_sep7a.ino: In function 'void setup()':
/private/var/folders/j0/_j5h8rw161l2lrx2khy0_2wh0000gn/T/.arduinoIDE-unsaved202387-31211-6pwcpw.qowm7/sketch_sep7a/sketch_sep7a.ino:39:9: error: 'class audio_tools::A2DPConfig' has no member named 'silence_on_nodata'
     cfg.silence_on_nodata = true; // allow delays with silence
         ^~~~~~~~~~~~~~~~~

exit status 1

Compilation error: 'class audio_tools::A2DPConfig' has no member named 'silence_on_nodata'
```
- Commenting line 39, enables me to compile but I just get a very fast "noise sound (seems it is trying to say a number)"

**Reply by pschatzmann** [reply]
You should have updated this project and not ESP32-A2DP!

**Reply by d3mac123** [reply]
I thought I needed to install the ESP32-A2DP library because you mentioned you had to "extend the A2DPStream...". Anyways, my bad.

So, besides creating a new sketch and pasting your example code, what else should I do? I am still not sure why I am getting the Compilation error: 'class audio_tools::A2DPConfig' has no member named 'silence_on_nodata' error.

**Reply by pschatzmann** [reply]
A2DPStream and resamping are all part of the AudioTools!
After all this issue is for the AudioTools as well.

**Reply by d3mac123** [reply]
So sorry, I am a dumbass! All working fine now. Again, really appreciate your help - you are the best!
